carbon/components/OptionsSwitcher/index.tsx

```diff
@@ -1,37 +1,78 @@
+"use client";
 /**
  * A widget to choose betwwen different options
+ * name: name of the widget (for state persistence state)
+ * options: Options offered by the widget
+ * onSelectionChange: function triggered when an option is selected
+ * value: initial value of the widget
+ * className: className to apply to the widget
  */
 import { OptionChangeHandler, Options } from "lib/charts/options";
-import { Key, useState } from "react";
+import { Key, useEffect, useState } from "react";
 import styles from "./styles.module.scss";
 
 export default function OptionsSwitcher<T extends Key>(props: {
+  name: string;
   options: Options<T>;
-  onSelectionChange: OptionChangeHandler<T>;
+  onSelectionChange?: OptionChangeHandler<T>;
   value?: T;
   className?: string;
+  readonly?: boolean;
 }) {
+  // The widget initial value is the first option value or the value from the url query
+  let initialValue = props.value || props.options[0].value;
+  if ("undefined" !== typeof window) {
+    const url = new URL(window.location.href);
+    initialValue = (url.searchParams.get(props.name) as T) || initialValue;
+  }
   const [selected, setSelected] = useState<T>(
     props.value || props.options[0].value
   );
+  const readonly = props.readonly === undefined ? false : props.readonly;
+
+  // Notify the parent of the initial value
+  useEffect(() => {
+    const url = new URL(window.location.href);
+    initialValue = (url.searchParams.get(props.name) as T) || initialValue;
+    setSelected(initialValue);
+    console.log(initialValue);
+    if (!readonly && props.onSelectionChange) {
+      props.onSelectionChange(initialValue);
+    }
+  }, []);
```

Comment:
We have similar code in `PageWithTabs/index.tsx` maybe we can extract that logic into a library function?

Reply:
done